go/enclave/events/subscription_manager.go

```diff
@@ -116,9 +116,16 @@ func (s *SubscriptionManager) FilteredLogs(logs []*types.Log, rollupHash common.
 	return filteredLogs
 }
 
-// FilteredSubscribedLogs filters out irrelevant logs, those that are not subscribed to, and those a subscription has
-// seen before, and organises them by their subscribing ID.
-func (s *SubscriptionManager) FilteredSubscribedLogs(logs []*types.Log, rollupHash common.L2RootHash) map[gethrpc.ID][]*types.Log {
+// EncryptFilteredLogs filters the logs based on the current subscriptions, then encrypts them the appropriate viewing
+// keys.
+func (s *SubscriptionManager) EncryptFilteredLogs(logs []*types.Log, rollupHash common.L2RootHash) (map[gethrpc.ID][]byte, error) {
+	filteredLogs := s.getFilteredLogs(logs, rollupHash)
+	return s.encryptLogs(filteredLogs)
+}
+
+// Filters out irrelevant logs, those that are not subscribed to, and those a subscription has seen before, and
+// organises them by their subscribing ID.
+func (s *SubscriptionManager) getFilteredLogs(logs []*types.Log, rollupHash common.L2RootHash) map[gethrpc.ID][]*types.Log {
```

Comment:
Similarly, this might be `getSubscribedLogs` or something, to make it clearer what it's filtering.